src/bot/modules/stats.ts

```diff
@@ -83,38 +93,85 @@ export default class Stats {
   }
 
   static async globalStats(interaction: CommandInteraction<'cached'>) {
-    const proposals = await prisma.proposal.groupBy({
-      by: ['user_id', 'merged'],
-      having: {
-        user_id: {
-          not: null
+    const proposals = await prisma.proposal.findMany({
+      include: {
+        _count: {
+          select: {
+            approvals: true,
+            disapprovals: true
+          }
         },
-        merged: true
-      },
-      _count: true
+        votes: true
+      }
     });
 
-    const membersProposals = proposals
-      .filter((proposal) => interaction.guild.members.cache.has(proposal.user_id!))
-      .sort((a, b) => b._count - a._count);
+    const membersProposals = proposals.filter(
+      (proposal) => proposal.user_id && interaction.guild.members.cache.has(proposal.user_id)
+    );
+    const membersVotes = proposals.reduce<Vote[]>((votes, proposal) => (votes.push(...proposal.votes), votes), []);
+    const godfathersDecisions = await Promise.all([prisma.approval.findMany(), prisma.disapproval.findMany()]);
+
+    const membersIds = [...new Set(membersProposals.map((p) => p.user_id!))];
+
+    const membersPoints = membersIds.map((userId) => {
+      const member = interaction.guild.members.cache.get(userId)!;
+
+      const memberProposals = membersProposals.filter((mP) => mP.user_id === userId);
+      const memberVotes = membersVotes.filter((mP) => mP.user_id === userId);
+
+      const points = this.calculatePoints(member, memberProposals, memberVotes, godfathersDecisions);
+
+      return `<@${userId}> : ${points}}`;
```

Comment:
```suggestion
    const membersPoints = membersIds.map(async (userId) => {
      const member = interaction.guild.members.cache.get(userId)!;

      const memberProposals = membersProposals.filter((p) => p.user_id === userId);
      const memberVotes = membersVotes.filter((v) => v.user_id === userId);

      const points = await this.calculatePoints(member, memberProposals, memberVotes, godfathersDecisions);

      return `<@${userId}> : ${points} ${points !== 1 ? 'points' : 'point'}`;
```

Reply:
Pour ca et si si cette fonction est async 
`This condition will always return 'true' since the types 'Promise<number>' and '1' have no overlap.`

Reply:
Ok je viens de modifier tu a laissez un async en trop a la fonction `calculatePoints`